pkg/client/cache/eventqueue.go

```diff
@@ -290,8 +290,16 @@ func (eq *EventQueue) Pop() (watch.EventType, interface{}, error) {
 		key := eq.queue[0]
 		eq.queue = eq.queue[1:]
 
-		eventType := eq.events[key]
-		delete(eq.events, key)
+		eventType, ok := eq.events[key]
+		if ok {
+			delete(eq.events, key)
+		} else {
+			// The event type has been removed by a previous call to
```

Comment:
Yes, it's possible that this is the only use of event queue in a meaningful form left in the code.